### PR TITLE
Added button and heading styles

### LINKED ISSUE #1: Button Styles
Need to define a base style for how our buttons will look, then need to add in contextual variants
E.g. 
Success button = green
Error button = red

### LINKED ISSUE #2: Typography
Need to define typography rules (fonts/sizes)

### LINKED ISSUE #3: UI Demo
There should be a page that allows the styling for every component/UI element to be easily visible

### PR DESCRIPTION
**Pull request by rugulous**
Added a few base styles:

- Default, Red and Green buttons (closes #1)
- h1-h6 and .h1-.h6 heading styles (closes #2)
- Added a "demo" site to allow previewing/checking for the styles (closes #3)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,34 @@
...
+The scripts in this directory are responsible for initialising any variables or mixins that may be required by the components
+
+3. components
+
+The scripts in this directory are responsible for creating the styling for each individual component and its states.
+Components are UI elements such as buttons or input fields, but can also be more complex things such as cards or modals.
+
+4. layout
+
+The scripts in this directory are responsible for initialising a grid system and styling the various different layut segments (e.g. container, header, footer)
```
Comment (matt-fidd) [inline README.md]: s/layut/layout

File: README.md [hunk 2]
```diff
@@ -1,2 +1,34 @@
 # Style
-Universal styling/branding
+Universal styling/branding using SCSS
+
+## Structure
+All the SCSS files required for compilation can be found in the scss directory.
+
+The entry point is `style.scss` - this should only every contain comments and imports. The imports will be made up of all the partial files, in the following order:
+
```
Comment (matt-fidd) [inline README.md:8]: Is scripts the right word, maybe stylesheets instead?

File: scss/style.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
+//sass modules
+@use 'sass:color';
+
+//reset styles
+@import 'base/_reset';
```
Comment (matt-fidd) [inline scss/style.scss]: You can omit the `_` when importing scss partials

File: scss/utils/functions/_math.scss [hunk 4]
```diff
@@ -0,0 +1,41 @@
+@function divide($dividend, $divisor, $precision: 10) {
```
Comment (matt-fidd) [inline scss/utils/functions/_math.scss:1]: The indentation seems odd in this one

File: scss/utils/mixins/_contrast.scss [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+@function color-contrast($background, $color-contrast-dark: $color-contrast-dark, $color-contrast-light: $color-contrast-light, $min-contrast-ratio: $min-contrast-ratio) {
```
Comment (matt-fidd) [inline scss/utils/mixins/_contrast.scss]: Ditto with the indentation here

**Comment by matt-fidd** [comment]
And I'd probably recommend that we don't ship the compiled css and map files, because these will be built when cloning anyway.

File: README.md (hunk 2 above)
Comment (rugulous) [reply README.md:8]: Will be honest, I'm not exactly sure what the right word is! I think stylesheets work for places that declare the actual style rules, but not sure if it's correct for mixins/functions? What do you think?

File: scss/style.scss (hunk 3 above)
Comment (rugulous) [reply scss/style.scss]: Didn't know that, thanks for the tip!

**Comment by rugulous** [comment]
Fair enough with not shipping the map.

I need the stylesheet *somewhere* here though so the demo page works (so we can see the styles easily) - unless there's a better way of doing this also?

File: scss/utils/functions/_math.scss (hunk 4 above)
Comment (rugulous) [reply scss/utils/functions/_math.scss:1]: Dodgy copy-paste, my bad...

**Comment by matt-fidd** [comment]
Would you not just build it when you clone the repo?

**Comment by rugulous** [comment]
@matt-fidd - could just be me being stupid.

At present there's a `/demo/` folder in the repo so we can see at a quick glance "here's how all the styles work" (I've spun it up here so you can see what I mean - [http://150.230.127.116:50000/demo/](http://150.230.127.116:50000/demo/))

It just means we're able to see what UI elements our framework allows for without having to have already cloned somewhere else and looking through the CSS.

It's worth debating whether this demo is actually needed/if it should be in this repo, but if it is staying here then the css file also needs to stay here so the demo can load it

File: README.md [hunk 6]
```diff
@@ -1,2 +1,34 @@
...
+The stylesheets in this directory are responsible for initialising a grid system and styling the various different layout segments (e.g. container, header, footer)
+
+
+The demo subdirectory contains the files necessary to view the components
+
+## Building
+Make sure you have sass installed (important!) then run
+
+```
+sass scss/style.scss style.scss
```
Comment (matt-fidd) [inline README.md:33]: Should this read as `sass scss/style.scss style.css` instead?

**Comment by matt-fidd** [comment]
I think the demo page is a great idea, but the css file for it can be built when the repo is downloaded manually as opposed to pushing a version into the repository.

The flow would be:
1. Clone the repo
2. Run `npx sass scss/style.scss style.css` (You could even make this an NPM script if you wanted, and have sass as a dependency)
3. Site works

**Comment by rugulous** [comment]
Fair enough, I've removed that from the repo too then!

**Comment by rugulous** [comment]
Will be implementing this in a different way

**Pull request closed without merging**